### PR TITLE
Add Creator/Custom Design Items

### PR DESCRIPTION
**Pull request by otaviocc**
## Description

There are so many creators and custom designs to track and try that sometimes it's hard to keep track of them. A good place to store these information would be in this companion app.

## Implementation

The main types added by this commit are:

- `Design`: simple model with title, code, and description.
- `DesignRowView` and `DesignRowViewModel`: view and view model used to populate the table.
- `DesignFormView` and `DesignFormViewModel`: view and view model used to add a new design to the list.

This commit also touches the caching layer, adding designs as an optional array.

## Screenshots

screenshots

## Demo

output

In case of bugs, I can help [Blathers](https://animalcrossing.fandom.com/wiki/Blathers) on that.

### CODE REVIEW

**Comment by Dimillian** [comment]
Hello, love the feature. But in the collection screen, this segmented control will crop the label on top. If you try on an iPhone 8 for example. Or a landscape iPad. Maybe it's not the right place to be or maybe we have to reword the top selector to something custom. I'll review the code shortly.

**Comment by otaviocc** [comment]
> Hello, love the feature. But in the collection screen, this segmented control will crop the label on top. If you try on an iPhone 8 for example. Or a landscape iPad. Maybe it's not the right place to be or maybe we have to rework the top selector to something custom. I'll review the code shortly.

Oops :-) Do you have any suggestion for the ui/ux, @Dimillian? I can give it a try.

**Comment by otaviocc** [comment]
> Hello, love the feature. But in the collection screen, this segmented control will crop the label on top. If you try on an iPhone 8 for example. Or a landscape iPad. Maybe it's not the right place to be or maybe we have to rework the top selector to something custom. I'll review the code shortly.

I have suggestion and will implement and propose it later tonight.

**Comment by Dimillian** [comment]
Yes sorry didn't had time to think about it yet. Can't wait to see your proposition.

**Comment by renaudjenny** [comment]
What about having a `More` tab which directs to a list of "other" available collections.
Screenshot 2020-05-26 at 18 46 20
Screenshot 2020-05-26 at 18 46 54

**Comment by Dimillian** [comment]
I like it. I would move critters into the more tab and leave the custom lists as it is. We already have a whole fritters section in the dashboard. And List have a subscribe call :p

**Comment by otaviocc** [comment]
new

I've added the "More" tab and moved Critters and Designs there. Let me know what you think, @Dimillian, @renaudjenny.

**Comment by Dimillian** [comment]
Awesome !!!! I guess this will land in master very soon :)

**Comment by renaudjenny** [comment]
@otaviocc awesome! Can't wait to have this on the app.
I quickly checked the code, and it's looking good!

File: ACHNBrowserUI/ACHNBrowserUI/views/collection/CollectionMoreDetailView.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+// MARK: - Preview
+
+#if DEBUG
+struct MoreView_Previews: PreviewProvider {
+    static var previews: some View {
+        List {
+            CollectionMoreDetailView(viewModel: CollectionMoreDetailViewModel())
+        }
+        .previewLayout(.sizeThatFits)
```
Comment (renaudjenny) [inline ACHNBrowserUI/ACHNBrowserUI/views/collection/CollectionMoreDetailView.swift:59]: That's smart. I've never think about this one!

**Comment by Dimillian** [comment]
Making few fixes for localization and merging, great works!

**Pull request merged**